File: java2csharp/java/Samsung/Camera/Sample_Filter.java
package com.samsung.android.sdk.camera.sample.cases;

import android.app.Activity;
import android.app.AlertDialog;
import android.content.DialogInterface;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.media.MediaScannerConnection;
import android.net.Uri;
import android.os.Bundle;
import android.os.Environment;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.samsung.android.sdk.SsdkUnsupportedException;
import com.samsung.android.sdk.camera.SCamera;
import com.samsung.android.sdk.camera.filter.SCameraFilter;
import com.samsung.android.sdk.camera.filter.SCameraFilterInfo;
import com.samsung.android.sdk.camera.filter.SCameraFilterManager;
import com.samsung.android.sdk.camera.sample.R;

import java.io.File;
import java.io.FileFilter;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.util.List;

public class Sample_Filter extends Activity implements View.OnClickListener {
    /**
     * Tag for the {@link Log}.
     */
    private static final String TAG = "Sample_Filter";

    // SCameraFilter only supports a 8192x8192 or less resolution.
    private static final int MAX_IMAGE_SIZE = 8192;

    private final int PROCESS_BITMAP = 1;
    private final int PROCESS_FILE = 2;

    private final String mCameraDirectory = Environment.getExternalStorageDirectory().getPath() + "/DCIM/Camera";

    private SCamera mSCamera = null;

    /**
     * list of retrieving available filters.
     */
    private List<SCameraFilterInfo> mFilterInfoList;

    /**
     * {@link com.samsung.android.sdk.camera.filter.SCameraFilterManager} for creating and retrieving available filters
     */
    private SCameraFilterManager mSCameraFilterManager;


    /**
     * Button to processing for bitmap format
     */
    private Button mBitmapButton = null;

    /**
     * Button to processing for image file
     */
    private Button mFileButton = null;

    /**
     * ImageView to display a selected image.
     */
    private ImageView mInputImageView = null;

    /**
     * ImageView to display a processed image.
     */
    private ImageView mOutputImageView = null;

    /**
     * TextView to display a selected filter.
     */
    private TextView mSelectedFilterTextView = null;

    /**
     * ID of selected filter.
     */
    private int mFilterID = -1;

    /**
     * list of filter name
     */
    private String[] mFilterNames = null;

    /**
     * list of file in [/DCIM/Camera]
     */
    private String[] mFileNames = null;

    /**
     * file path of a processed image.
     */
    private String mInputImage = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_filter);

        mSelectedFilterTextView = (TextView) findViewById(R.id.selected_textview);

        findViewById(R.id.btn_select_image).setOnClickListener(this);
        findViewById(R.id.btn_select_filter).setOnClickListener(this);

        mBitmapButton = (Button) findViewById(R.id.btn_bitmap);
        mFileButton = (Button) findViewById(R.id.btn_file);

        mBitmapButton.setOnClickListener(this);
        mFileButton.setOnClickListener(this);

        mBitmapButton.setEnabled(false);
        mFileButton.setEnabled(false);

        mInputImageView = (ImageView) findViewById(R.id.input_image_view);
        mOutputImageView = (ImageView) findViewById(R.id.output_image_view);

        mSCamera = new SCamera();
        try {
            mSCamera.initialize(this);
        } catch (SsdkUnsupportedException e) {
            showAlertDialog("Fail to initialize SCamera.", true);
            return;
        }
        if (!mSCamera.isFeatureEnabled(SCamera.SCAMERA_FILTER)) {
            showAlertDialog("This device does not support SCamera Filter feature.", true);
            return;
        }

        // retrieving an {@link com.samsung.android.sdk.camera.filter.SCameraFilterManager}
        mSCameraFilterManager = mSCamera.getSCameraFilterManager();

        // retrieving available filters
        mFilterInfoList = mSCameraFilterManager.getAvailableFilters();

        if (mFilterInfoList.size() > 0)
            mSelectedFilterTextView.setText(mFilterInfoList.get(0).getName());

    }

    @Override
    protected void onDestroy() {
        mSCamera = null;
        super.onDestroy();
    }

    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.btn_select_image:
                selectImage();
                break;
            case R.id.btn_select_filter:
                selectFilter();
                break;
            case R.id.btn_bitmap:
                process(PROCESS_BITMAP);
                break;
            case R.id.btn_file:
                process(PROCESS_FILE);
                break;
        }
    }

    /**
     * select an image file for processing filter.
     */
    private void selectImage() {

        mFileNames = null;

        File cameraDirectory = new File(mCameraDirectory);
        if (!cameraDirectory.exists()) {
            cameraDirectory.mkdirs();
            return;
        }

        File[] cameraImages = cameraDirectory.listFiles(new FileFilter() {
            public boolean accept(File file) {
                return file.isFile() && file.getName().toLowerCase().endsWith(".jpg");
            }
        });
        if (cameraImages == null || cameraImages.length == 0) {
            return;
        }

        mFileNames = new String[cameraImages.length];

        for (int i = 0; i < cameraImages.length; i++) {
            mFileNames[i] = mCameraDirectory + "/" + cameraImages[i].getName();
        }

        new AlertDialog.Builder(Sample_Filter.this)
                .setTitle("Select Image")
                .setItems(mFileNames, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        String inputImage = mFileNames[which];

                        if (!setImageView(mInputImageView, inputImage)) {
                            showAlertDialog("This image resolution is greater than the " + MAX_IMAGE_SIZE + "x" + MAX_IMAGE_SIZE, false);
                            return;
                        }

                        mInputImage = inputImage;
                        mBitmapButton.setEnabled(true);
                        mFileButton.setEnabled(true);
                    }
                }).show();

    }

    @Override
    protected void onResume() {
        super.onResume();

        if(mInputImage != null) {
            File inputFile = new File(mInputImage);
            if(!inputFile.exists()) {
                mInputImage = null;
                mBitmapButton.setEnabled(false);
                mFileButton.setEnabled(false);
                mInputImageView.setImageDrawable(getDrawable(R.drawable.bg_image_frame));
                mOutputImageView.setImageDrawable(getDrawable(R.drawable.bg_image_frame));
            }
        }
    }

    /**
     * select an filter file for processing.
     */
    private void selectFilter() {

        int filterNum = mFilterInfoList.size();
        if (filterNum < 1)
            return;

        mFilterNames = new String[filterNum];

        for (int i = 0; i < filterNum; i++) {
            mFilterNames[i] = mFilterInfoList.get(i).getName();
        }

        new AlertDialog.Builder(Sample_Filter.this)
                .setTitle("Select Filter")
                .setItems(mFilterNames, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        mFilterID = which;
                        mSelectedFilterTextView.setText(mFilterNames[which]);
                    }
                }).show();
    }

    private void process(int type) {
        if (mInputImage == null) {
            Toast tempToast = Toast.makeText(Sample_Filter.this, "please select image.", Toast.LENGTH_SHORT);
            tempToast.show();
            return;
        }

        String outputPath = mCameraDirectory + "/FILTER_IMAGE.jpg";
        processFilter(mInputImage, outputPath, type);
        setImageView(mOutputImageView, outputPath);

        MediaScannerConnection.scanFile(this,
                new String[]{outputPath}, null,
                new MediaScannerConnection.OnScanCompletedListener() {
                    public void onScanCompleted(String path, Uri uri) {
                        Log.i(TAG, "ExternalStorage Scanned " + path + "-> uri=" + uri);
                    }
                });
    }

    /**
     * display an image file.
     */
    private boolean setImageView(ImageView imageview, String filepath) {

        boolean bSCameraFilterSupported = true;

        BitmapFactory.Options options = new BitmapFactory.Options();
        options.inJustDecodeBounds = true;
        BitmapFactory.decodeFile(filepath, options);

        int photoWidth = options.outWidth;
        int photoHeight = options.outHeight;

        int targetWidth = photoWidth;
        int targetHeight = photoHeight;

        int photoSize = (photoWidth > photoHeight) ? photoWidth : photoHeight;

        if (photoSize> MAX_IMAGE_SIZE) {
            bSCameraFilterSupported = false;
            return bSCameraFilterSupported;
        }

        if (photoSize > 1920) {
            int scale = (photoSize / 1920) + 1;
            targetWidth = photoWidth / scale;
            targetHeight = photoHeight / scale;
        }

        int scaleFactor = Math.min(photoWidth / targetWidth, photoHeight / targetHeight);

        options.inJustDecodeBounds = false;
        options.inDither = false;
        options.inSampleSize = scaleFactor;
        Bitmap orgImage = BitmapFactory.decodeFile(filepath, options);
        imageview.setImageBitmap(orgImage);

        return bSCameraFilterSupported;
    }

    /**
     * save an bitmap object.
     */
    private void saveImage(Bitmap bitmap, String filename) {

        if (bitmap != null) {

            File file = new File(filename);
            OutputStream out = null;
            try {
                file.createNewFile();
                out = new FileOutputStream(file);
                bitmap.compress(Bitmap.CompressFormat.JPEG, 95, out);
                bitmap.recycle();
            } catch (Exception e) {
                e.printStackTrace();
            } finally {
                if(out != null) {
                    try {
                    out.close();
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                }
            }

        }
    }

    // ----------------------------------------------------------------------------

    /**
     * process a filter
     */
    private void processFilter(String inputFile, String outputFile, int type) {
        Log.e(TAG, "InputFile:  " + inputFile);
        Log.e(TAG, "OutputFile:  " + outputFile);
        Log.e(TAG, "Type:" + type);
        Log.e(TAG, "Filter:  " + mFilterID);

        SCameraFilter filter;

        // create a SCameraFilter using SCameraFilterManager.
        if (mFilterID == -1)
            filter = mSCameraFilterManager.createFilter(mFilterInfoList.get(0));
        else
            filter = mSCameraFilterManager.createFilter(mFilterInfoList.get(mFilterID));

        switch (type) {
            case PROCESS_BITMAP:
                Bitmap inputBitmap = BitmapFactory.decodeFile(inputFile);

                // process filter using bitmap object
                Bitmap outputBitmap = filter.processImage(inputBitmap);
                if (outputBitmap == null) {
                    return;
                }

                // save a bitmap object
                saveImage(outputBitmap, outputFile);

                inputBitmap.recycle();
                outputBitmap.recycle();
                break;
            case PROCESS_FILE:
                // process filter using image file
                filter.processImage(inputFile, outputFile);
                break;
        }
    }

    /**
     * Shows alert dialog.
     */
    private void showAlertDialog(String message, final boolean finishActivity) {

        final AlertDialog.Builder dialog = new AlertDialog.Builder(this);
        dialog.setMessage(message)
            .setIcon(android.R.drawable.ic_dialog_alert)
            .setTitle("Alert")
            .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    dialog.dismiss();
                    if (finishActivity) finish();
                }
            }).setCancelable(false);

        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                dialog.show();
            }
        });
    }
}

File: java2csharp/csharp/Samsung/Camera/cases/Sample_Filter.cs
using System;
using System.Collections.Generic;

namespace com.samsung.android.sdk.camera.sample.cases
{

	using Activity = android.app.Activity;
	using AlertDialog = android.app.AlertDialog;
	using DialogInterface = android.content.DialogInterface;
	using Bitmap = android.graphics.Bitmap;
	using BitmapFactory = android.graphics.BitmapFactory;
	using MediaScannerConnection = android.media.MediaScannerConnection;
	using Uri = android.net.Uri;
	using Bundle = android.os.Bundle;
	using Environment = android.os.Environment;
	using Log = android.util.Log;
	using View = android.view.View;
	using Button = android.widget.Button;
	using ImageView = android.widget.ImageView;
	using TextView = android.widget.TextView;
	using Toast = android.widget.Toast;

	using SCameraFilter = com.samsung.android.sdk.camera.filter.SCameraFilter;
	using SCameraFilterInfo = com.samsung.android.sdk.camera.filter.SCameraFilterInfo;
	using SCameraFilterManager = com.samsung.android.sdk.camera.filter.SCameraFilterManager;


	public class Sample_Filter : Activity, View.OnClickListener
	{
		/// <summary>
		/// Tag for the <seealso cref="Log"/>.
		/// </summary>
		private const string TAG = "Sample_Filter";

		// SCameraFilter only supports a 8192x8192 or less resolution.
		private const int MAX_IMAGE_SIZE = 8192;

		private readonly int PROCESS_BITMAP = 1;
		private readonly int PROCESS_FILE = 2;

		private readonly string mCameraDirectory = Environment.ExternalStorageDirectory.Path + "/DCIM/Camera";

		private SCamera mSCamera = null;

		/// <summary>
		/// list of retrieving available filters.
		/// </summary>
		private IList<SCameraFilterInfo> mFilterInfoList;

		/// <summary>
		/// <seealso cref="com.samsung.android.sdk.camera.filter.SCameraFilterManager"/> for creating and retrieving available filters
		/// </summary>
		private SCameraFilterManager mSCameraFilterManager;


		/// <summary>
		/// Button to processing for bitmap format
		/// </summary>
		private Button mBitmapButton = null;

		/// <summary>
		/// Button to processing for image file
		/// </summary>
		private Button mFileButton = null;

		/// <summary>
		/// ImageView to display a selected image.
		/// </summary>
		private ImageView mInputImageView = null;

		/// <summary>
		/// ImageView to display a processed image.
		/// </summary>
		private ImageView mOutputImageView = null;

		/// <summary>
		/// TextView to display a selected filter.
		/// </summary>
		private TextView mSelectedFilterTextView = null;

		/// <summary>
		/// ID of selected filter.
		/// </summary>
		private int mFilterID = -1;

		/// <summary>
		/// list of filter name
		/// </summary>
		private string[] mFilterNames = null;

		/// <summary>
		/// list of file in [/DCIM/Camera]
		/// </summary>
		private string[] mFileNames = null;

		/// <summary>
		/// file path of a processed image.
		/// </summary>
		private string mInputImage = null;

		protected internal override void onCreate(Bundle savedInstanceState)
		{
			base.onCreate(savedInstanceState);

			ContentView = R.layout.activity_filter;

			mSelectedFilterTextView = (TextView) findViewById(R.id.selected_textview);

			findViewById(R.id.btn_select_image).OnClickListener = this;
			findViewById(R.id.btn_select_filter).OnClickListener = this;

			mBitmapButton = (Button) findViewById(R.id.btn_bitmap);
			mFileButton = (Button) findViewById(R.id.btn_file);

			mBitmapButton.OnClickListener = this;
			mFileButton.OnClickListener = this;

			mBitmapButton.Enabled = false;
			mFileButton.Enabled = false;

			mInputImageView = (ImageView) findViewById(R.id.input_image_view);
			mOutputImageView = (ImageView) findViewById(R.id.output_image_view);

			mSCamera = new SCamera();
			try
			{
				mSCamera.initialize(this);
			}
			catch (SsdkUnsupportedException)
			{
				showAlertDialog("Fail to initialize SCamera.", true);
				return;
			}
			if (!mSCamera.isFeatureEnabled(SCamera.SCAMERA_FILTER))
			{
				showAlertDialog("This device does not support SCamera Filter feature.", true);
				return;
			}

			// retrieving an {@link com.samsung.android.sdk.camera.filter.SCameraFilterManager}
			mSCameraFilterManager = mSCamera.SCameraFilterManager;

			// retrieving available filters
			mFilterInfoList = mSCameraFilterManager.AvailableFilters;

			if (mFilterInfoList.Count > 0)
			{
				mSelectedFilterTextView.Text = mFilterInfoList[0].Name;
			}

		}

		protected internal override void onDestroy()
		{
			mSCamera = null;
			base.onDestroy();
		}

		public virtual void onClick(View v)
		{
			switch (v.Id)
			{
				case R.id.btn_select_image:
					selectImage();
					break;
				case R.id.btn_select_filter:
					selectFilter();
					break;
				case R.id.btn_bitmap:
					process(PROCESS_BITMAP);
					break;
				case R.id.btn_file:
					process(PROCESS_FILE);
					break;
			}
		}

		/// <summary>
		/// select an image file for processing filter.
		/// </summary>
		private void selectImage()
		{

			mFileNames = null;

			File cameraDirectory = new File(mCameraDirectory);
			if (!cameraDirectory.exists())
			{
				cameraDirectory.mkdirs();
				return;
			}

			File[] cameraImages = cameraDirectory.listFiles(new FileFilterAnonymousInnerClassHelper(this));
			if (cameraImages == null || cameraImages.Length == 0)
			{
				return;
			}

			mFileNames = new string[cameraImages.Length];

			for (int i = 0; i < cameraImages.Length; i++)
			{
				mFileNames[i] = mCameraDirectory + "/" + cameraImages[i].Name;
			}

			(new AlertDialog.Builder(Sample_Filter.this)).setTitle("Select Image").setItems(mFileNames, new OnClickListenerAnonymousInnerClassHelper(this))
				   .show();

		}

		private class FileFilterAnonymousInnerClassHelper : FileFilter
		{
			private readonly Sample_Filter outerInstance;

			public FileFilterAnonymousInnerClassHelper(Sample_Filter outerInstance)
			{
				this.outerInstance = outerInstance;
			}

			public virtual bool accept(File file)
			{
				return file.File && file.Name.toLowerCase().EndsWith(".jpg", StringComparison.Ordinal);
			}
		}

		private class OnClickListenerAnonymousInnerClassHelper : DialogInterface.OnClickListener
		{
			private readonly Sample_Filter outerInstance;

			public OnClickListenerAnonymousInnerClassHelper(Sample_Filter outerInstance)
			{
				this.outerInstance = outerInstance;
			}

			public override void onClick(DialogInterface dialog, int which)
			{
				string inputImage = outerInstance.mFileNames[which];

				if (!outerInstance.setImageView(outerInstance.mInputImageView, inputImage))
				{
					outerInstance.showAlertDialog("This image resolution is greater than the " + MAX_IMAGE_SIZE + "x" + MAX_IMAGE_SIZE, false);
					return;
				}

				outerInstance.mInputImage = inputImage;
				outerInstance.mBitmapButton.Enabled = true;
				outerInstance.mFileButton.Enabled = true;
			}
		}

		protected internal override void onResume()
		{
			base.onResume();

			if (mInputImage != null)
			{
				File inputFile = new File(mInputImage);
				if (!inputFile.exists())
				{
					mInputImage = null;
					mBitmapButton.Enabled = false;
					mFileButton.Enabled = false;
					mInputImageView.ImageDrawable = getDrawable(R.drawable.bg_image_frame);
					mOutputImageView.ImageDrawable = getDrawable(R.drawable.bg_image_frame);
				}
			}
		}

		/// <summary>
		/// select an filter file for processing.
		/// </summary>
		private void selectFilter()
		{

			int filterNum = mFilterInfoList.Count;
			if (filterNum < 1)
			{
				return;
			}

			mFilterNames = new string[filterNum];

			for (int i = 0; i < filterNum; i++)
			{
				mFilterNames[i] = mFilterInfoList[i].Name;
			}

			(new AlertDialog.Builder(Sample_Filter.this)).setTitle("Select Filter").setItems(mFilterNames, new OnClickListenerAnonymousInnerClassHelper2(this))
				   .show();
		}

		private class OnClickListenerAnonymousInnerClassHelper2 : DialogInterface.OnClickListener
		{
			private readonly Sample_Filter outerInstance;

			public OnClickListenerAnonymousInnerClassHelper2(Sample_Filter outerInstance)
			{
				this.outerInstance = outerInstance;
			}

			public override void onClick(DialogInterface dialog, int which)
			{
				outerInstance.mFilterID = which;
				outerInstance.mSelectedFilterTextView.Text = outerInstance.mFilterNames[which];
			}
		}

		private void process(int type)
		{
			if (mInputImage == null)
			{
				Toast tempToast = Toast.makeText(Sample_Filter.this, "please select image.", Toast.LENGTH_SHORT);
				tempToast.show();
				return;
			}

			string outputPath = mCameraDirectory + "/FILTER_IMAGE.jpg";
			processFilter(mInputImage, outputPath, type);
			setImageView(mOutputImageView, outputPath);

			MediaScannerConnection.scanFile(this, new string[]{outputPath}, null, new OnScanCompletedListenerAnonymousInnerClassHelper(this));
		}

		private class OnScanCompletedListenerAnonymousInnerClassHelper : MediaScannerConnection.OnScanCompletedListener
		{
			private readonly Sample_Filter outerInstance;

			public OnScanCompletedListenerAnonymousInnerClassHelper(Sample_Filter outerInstance)
			{
				this.outerInstance = outerInstance;
			}

			public virtual void onScanCompleted(string path, Uri uri)
			{
				Log.i(TAG, "ExternalStorage Scanned " + path + "-> uri=" + uri);
			}
		}

		/// <summary>
		/// display an image file.
		/// </summary>
		private bool setImageView(ImageView imageview, string filepath)
		{

			bool bSCameraFilterSupported = true;

			BitmapFactory.Options options = new BitmapFactory.Options();
			options.inJustDecodeBounds = true;
			BitmapFactory.decodeFile(filepath, options);

			int photoWidth = options.outWidth;
			int photoHeight = options.outHeight;

			int targetWidth = photoWidth;
			int targetHeight = photoHeight;

			int photoSize = (photoWidth > photoHeight) ? photoWidth : photoHeight;

			if (photoSize > MAX_IMAGE_SIZE)
			{
				bSCameraFilterSupported = false;
				return bSCameraFilterSupported;
			}

			if (photoSize > 1920)
			{
				int scale = (photoSize / 1920) + 1;
				targetWidth = photoWidth / scale;
				targetHeight = photoHeight / scale;
			}

			int scaleFactor = Math.Min(photoWidth / targetWidth, photoHeight / targetHeight);

			options.inJustDecodeBounds = false;
			options.inDither = false;
			options.inSampleSize = scaleFactor;
			Bitmap orgImage = BitmapFactory.decodeFile(filepath, options);
			imageview.ImageBitmap = orgImage;

			return bSCameraFilterSupported;
		}

		/// <summary>
		/// save an bitmap object.
		/// </summary>
		private void saveImage(Bitmap bitmap, string filename)
		{

			if (bitmap != null)
			{

				File file = new File(filename);
				System.IO.Stream @out = null;
				try
				{
					file.createNewFile();
					@out = new System.IO.FileStream(file, System.IO.FileMode.Create, System.IO.FileAccess.Write);
					bitmap.compress(Bitmap.CompressFormat.JPEG, 95, @out);
					bitmap.recycle();
				}
				catch (Exception e)
				{
					Console.WriteLine(e.ToString());
					Console.Write(e.StackTrace);
				}
				finally
				{
					if (@out != null)
					{
						try
						{
						@out.Close();
						}
						catch (IOException e)
						{
							Console.WriteLine(e.ToString());
							Console.Write(e.StackTrace);
						}
					}
				}

			}
		}

		// ----------------------------------------------------------------------------

		/// <summary>
		/// process a filter
		/// </summary>
		private void processFilter(string inputFile, string outputFile, int type)
		{
			Log.e(TAG, "InputFile:  " + inputFile);
			Log.e(TAG, "OutputFile:  " + outputFile);
			Log.e(TAG, "Type:" + type);
			Log.e(TAG, "Filter:  " + mFilterID);

			SCameraFilter filter;

			// create a SCameraFilter using SCameraFilterManager.
			if (mFilterID == -1)
			{
				filter = mSCameraFilterManager.createFilter(mFilterInfoList[0]);
			}
			else
			{
				filter = mSCameraFilterManager.createFilter(mFilterInfoList[mFilterID]);
			}

			switch (type)
			{
				case PROCESS_BITMAP:
					Bitmap inputBitmap = BitmapFactory.decodeFile(inputFile);

					// process filter using bitmap object
					Bitmap outputBitmap = filter.processImage(inputBitmap);
					if (outputBitmap == null)
					{
						return;
					}

					// save a bitmap object
					saveImage(outputBitmap, outputFile);

					inputBitmap.recycle();
					outputBitmap.recycle();
					break;
				case PROCESS_FILE:
					// process filter using image file
					filter.processImage(inputFile, outputFile);
					break;
			}
		}

		/// <summary>
		/// Shows alert dialog.
		/// </summary>
//JAVA TO C# CONVERTER WARNING: 'final' parameters are not available in .NET:
//ORIGINAL LINE: private void showAlertDialog(String message, final boolean finishActivity)
		private void showAlertDialog(string message, bool finishActivity)
		{

//JAVA TO C# CONVERTER WARNING: The original Java variable was marked 'final':
//ORIGINAL LINE: final android.app.AlertDialog.Builder dialog = new android.app.AlertDialog.Builder(this);
			AlertDialog.Builder dialog = new AlertDialog.Builder(this);
			dialog.setMessage(message).setIcon(android.R.drawable.ic_dialog_alert).setTitle("Alert").setPositiveButton(android.R.@string.ok, new OnClickListenerAnonymousInnerClassHelper3(this, finishActivity, dialog))
			   .Cancelable = false;

			runOnUiThread(() =>
			{
				dialog.show();
			});
		}

		private class OnClickListenerAnonymousInnerClassHelper3 : DialogInterface.OnClickListener
		{
			private readonly Sample_Filter outerInstance;

			private bool finishActivity;
			private AlertDialog.Builder dialog;

			public OnClickListenerAnonymousInnerClassHelper3(Sample_Filter outerInstance, bool finishActivity, AlertDialog.Builder dialog)
			{
				this.outerInstance = outerInstance;
				this.finishActivity = finishActivity;
				this.dialog = dialog;
			}

			public override void onClick(DialogInterface dialog, int which)
			{
				dialog.dismiss();
				if (finishActivity)
				{
					finish();
				}
			}
		}
	}

}